File: UniversityAPI/Entities/DTO/DeptRemoveVM.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Entities.DTO
{
    // faculty depends on department
    // course depends on faculty
    // while removing department, check for faculty and course dependancy
    public class DeptRemoveVM
    {
        public int DepartmentId { get; set; }
        public string DepartmentName { get; set; }
        public int ErrorCode { get; set; }
        public string ErrorMessage { get; set; }
        public List<FacultyRemoveVM> DependingFaculties { get; set; }
        public List<CourseRemoveVM> DependingCourses { get; set; }

    }
    public class FacultyRemoveVM
    {
        public int FacultyId { get; set; }
        public string Name { get; set; }
    }
    public class CourseRemoveVM
    {
        public int CourseId { get; set; }
        public string Name { get; set; }
    }
}

File: ngApp/src/app/models/deptRemoveVM.ts
import AssignmentRemoveVM from "./AssignmentRemoveVM";
import CourseRemoveVM from "./courseRemoveVM";
import FacultyRemoveVM from "./facultyRemoveVM";

export default class DeptRemoveVM {
    departmentId: number;
    departmentName: string;
    errorCode: number;
    errorMessage: string;
    dependingFaculties: FacultyRemoveVM[];
    dependingCourses: CourseRemoveVM[];    
}
